src/repository/couch/CredentialRepository.ts

```diff
@@ -32,7 +33,9 @@ export class CredentialRepository implements CredentialInterface {
       credential.origin = type_id
     }
     // Verify this is "our" credential correctly
-    if (credential.origin !== type_id || !credential.access_key || !credential.secret_key)
+    if (credential.origin !== type_id ||
+      !credential.access_key ||
+      !validateSecretKey(credential.secret_key))
```

Comment:
This is not only validating the secret key, but the entire credential object for both modes. I think if we wanted to go ahead with this change it would have to be as a part of a wider discussion around error handling in the backend as whole.

For example, the issue could be the access key and in that case the error message would make things confusing. I think the current one is a bit more generic and fits with the catch all nature of the validation.

Reply:
Ah, that makes sense. Thank you for explaining!